src/main/java/org/folio/service/transactions/batch/BatchTransactionService.java

```diff
@@ -0,0 +1,229 @@
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ALLOCATION;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.CREDIT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ENCUMBRANCE;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.PAYMENT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.PENDING_PAYMENT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.TRANSFER;
+
+public class BatchTransactionService {
+  private static final Logger logger = LogManager.getLogger();
+  private static final List<TransactionType> transactionTypesInOrder = List.of(ALLOCATION, TRANSFER, PAYMENT,
+    PENDING_PAYMENT, ENCUMBRANCE);
+  private final DBClientFactory dbClientFactory;
+  private final BatchTransactionDAO transactionDAO;
+  private final FundService fundService;
+  private final BudgetService budgetService;
+  private final LedgerService ledgerService;
+  private final Map<TransactionType, BatchTransactionServiceInterface> serviceMap;
+
+  public BatchTransactionService(DBClientFactory dbClientFactory, BatchTransactionDAO transactionDAO,
+      FundService fundService, BudgetService budgetService, LedgerService ledgerService,
+      Set<BatchTransactionServiceInterface> batchTransactionStrategies) {
+    this.dbClientFactory = dbClientFactory;
+    this.transactionDAO = transactionDAO;
+    this.fundService = fundService;
+    this.budgetService = budgetService;
+    this.ledgerService = ledgerService;
+    serviceMap = new EnumMap<>(TransactionType.class);
+    batchTransactionStrategies.forEach(
+      strategy -> serviceMap.put(strategy.getTransactionType(), strategy));
+  }
+
+  public Future<Void> processBatch(Batch batch, RequestContext requestContext) {
+    try {
+      BatchTransactionChecks.sanityChecks(batch);
+    } catch (Exception ex) {
+      logger.error(ex);
+      return Future.failedFuture(ex);
+    }
+    populateMetadata(batch, requestContext.getHeaders());
+    DBClient client = dbClientFactory.getDbClient(requestContext);
+    BatchTransactionHolder holder = new BatchTransactionHolder(transactionDAO, fundService, budgetService, ledgerService);
+    return client.withTrans(conn -> holder.setup(batch, conn)
+      .map(v -> {
+        BatchTransactionChecks.checkBudgetsAreActive(holder);
+        updatesForCreatingTransactions(holder);
+        updatesForUpdatingTransactions(holder);
+        updatesForPatchingTransactions(holder);
+        BatchTransactionChecks.checkRestrictedBudgets(holder);
+        return null;
+      })
+      .compose(v -> applyChanges(holder, conn))
+    ).onSuccess(v -> logger.info("All batch transaction operations were successful."))
+      .onFailure(t -> logger.error("Error when batch processing transactions, batch={}",
+        JsonObject.mapFrom(batch).encodePrettily(), t));
+  }
+
+  private void updatesForCreatingTransactions(BatchTransactionHolder holder) {
```

Comment:
let's use the same naming convention here: prepareCreatingTransactions(), prepareUpdatingTransactions()

Reply:
OK